src/test/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilderTest.java

```diff
@@ -595,8 +595,8 @@ public void given__cloud_branch_rev_userkey_sshtrait_userkey__when__build__then_
         createGitHubSCMSourceForTest(false, null);
         BranchSCMHead head = new BranchSCMHead("test-branch");
         SCMRevisionImpl revision = new SCMRevisionImpl(head, "cafebabedeadbeefcafebabedeadbeefcafebabe");
-        source.setCredentialsId("user-key");
         GitHubSCMBuilder instance = new GitHubSCMBuilder(source, head, revision);
+        instance.withCredentials("user-key", GitHubSCMBuilder.SSH);
```

Comment:
Nit: prefer non deprecated API usage
```suggestion
        GitHubSCMBuilder instance = new GitHubSCMBuilder(source, head, revision)
                .withCredentials("user-key")
                .withResolver(GitHubSCMBuilder.SSH);
```